M2/Macaulay2/packages/SimpleDoc/doc.txt

```diff
@@ -12,7 +29,7 @@ Node
  Consequences
   Item
    The string {\tt s} is handled by parsing it the same way @ TO doc @ does,
-   but expecting the least indented keyword to be {\bf Node}, which can occur
+   but expecting the least indented keyword to be @ TO Node @, which can occur
```

Comment:
There's no reason for things such as "Node" to be exported symbols, since they occur only inside strings and not in Macaulay2 code.

Reply:
It is a keyword in the documentation system, so I think it should be highlighted so users know it's correct. Is there another, easy way to highlight it?

How about we change `Node` to `doc`? Then the format would be:
```
doc
  Key
    blah
  Headline
    blah blah
doc
  Key
    Foo
  Headline
    Foo Foo
```
It is already highlighted, and semantically it makes sense that the input to `multidoc` is made up of a bunch of inputs to `doc`. Also, if you have a documentation file with a bunch of calls to `doc`, you can change it to a text file for `multidoc` just by removing `///` everywhere.

Reply:
> It is a keyword in the documentation system, so I think it should be highlighted so users know it's correct. Is there another easy way to highlight it?

Yes, there is: just add them to the list in `emacs/makesyms.m2`.

Reply:
What do you think about changing `Node` to `doc`? There are only about 20 packages that use it, so changing it wouldn't be difficult.

Reply:
> What do you think about changing `Node` to `doc`? There are only about 20 packages that use it, so changing it wouldn't be difficult.

I see the following disadvantages:

- "doc" is not a word
- "doc" is not capitalized
- "doc" is already in use for something else in the package
- the word "Node" is descriptive: it introduces a documentation node
- the change would not be backward compatible.

Here's a change I would like to see instead, even though it's backward compatible, if @mikestillman approves:

- get rid of "doc" and "multidoc"
- attach the functionality of "multidoc" as the method for "document String", which is currently empty

It would also be cool to develop an emacs mode for SimpleDoc files.  It could take care of the indentation and helping us know which keywords are valid at the current level of indentation.

Reply:
I like that. What about also moving the existing `document` method into SimpleDoc?

Reply:
`document` is part of the core, so should stay there.